oak_remote_attestation/proto/v1/evidence.proto

```diff
@@ -0,0 +1,119 @@
+}
+
+message RootLayerEvidence {
+  // The platform providing the attestation report.
+  TeePlatform platform = 0;
+  // TEE-specific attestation report acting as a non-standard certificate for the
+  // Layer0 ECA public key.
+  bytes remote_attestation_report = 1;
+  // Serialised ECA public key for Layer0.
+  bytes eca_public_key = 2;
+}
+
+message LayerEvidence {
+  string layer_name = 0;
+  // CBOR/COSE/CWT ECA certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  bytes eca_certificate = 1;
+}
+
+message ApplicationKeys {
+  // CBOR/COSE/CWT encryption public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes encryption_public_key_certificate = 0;
+  // CBOR/COSE/CWT signing public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes signing_public_key_certificate = 1;
+}
+
+message Evidence {
+  // Layer0 attestation evidence.
+  RootLayerEvidence root_layer_evidence = 0;
+  // Layer1..LayerN-1 attestation evidence.
+  repeated LayerEvidence layer_evidence = 1;
+  // Final layer keys.
+  optional ApplicationKeys application_keys = 2;
+}
+
+// Endorsement contains statements that some entity (e.g., a hardware provider) vouches for the
+// integrity of claims about the TEE or the software running on it.
+// The name is chosen to match the RATS terminology:
+// <https://www.rfc-editor.org/rfc/rfc9334.html#name-endorsements>
+message Endorsement {
+  // The serialized TEE certificate(s). The details of the format and how the certificate(s) are
+  // encoded into this byte array are implementation-specific. In case of AMD-SEV-SNP, as described
+  // in https://www.amd.com/system/files/TechDocs/57230.pdf, there are three different certificates
+  // packaged in two different files.
+  bytes tee_certificates = 1;
+
+  // Transparent Release endorsement.
+  TransparentReleaseEndorsement transparent_release_endorsement = 2;
+}
+
+message TransparentReleaseEndorsement {
+  // The serialized binary endorsement statement from the transparent release process.
+  bytes endorsement_statement = 1;
+  // The serialized Rekor LogEntry as proof of the inclusion of the endorsement statement in Rekor.
+  bytes rekor_log_entry = 2;
+  // Public key of rekor at the time of uploading the endorsement statement. This is required for
+  // verifying the log entry.
+  string base64_pem_encoded_rekor_public_key = 3;
+}
+
+// EndorsedEvidence contains the information that the untrusted launcher provides to the client
+// in response to its request for the enclave's public key(s).
+message EndorsedEvidence {
+  // Attestation evidence from the enclave.
+  Evidence evidence = 1;
+
+  // Supporting evidence required for verifying the integrity of attestation evidence.
+  Endorsement endorsement = 2;
+}
+
+// Message for passing embedded certificate authoricy information between layers.
+// Will never appear in the evidence that is sent to the client.
+message CertificateAuthority {
```

Comment:
If this is something internal to the DICE implementation that the client will never need, perhaps this should go into a separate file and package.